packages/@sanity/form-builder/src/inputs/PortableText/object/BlockObject.tsx

```diff
@@ -0,0 +1,327 @@
+      }
+    }
+
+    &[data-warning] {
+      &:after {
+        ${overlay}
+        background-color: ${color.muted.caution.hovered.bg};
+      }
+
+      @media (hover: hover) {
+        &:hover {
+          --card-border-color: ${color.muted.caution.hovered.border};
+        }
+      }
+    }
+
+    &[data-invalid] {
+      &:after {
+        ${overlay}
+        background-color: ${color.input.invalid.enabled.bg};
+      }
+
+      @media (hover: hover) {
+        &:hover {
+          --card-border-color: ${color.input.invalid.hovered.border};
+        }
+      }
+    }
+  `
+})
+
+const ChangeIndicatorWrapper = styled.div(({theme}: {theme: Theme}) => {
+  const {space} = theme.sanity
+
+  return css`
+    position: absolute;
+    width: ${space[2]}px;
+    right: -${space[2]}px;
+    top: -${space[1]}px;
+    bottom: -${space[1]}px;
+    overflow-x: hidden;
+    padding-left: ${space[1]}px;
+
+    [data-dragged] & {
+      visibility: hidden;
+    }
+  `
+})
+
+const StyledChangeIndicatorWithProvidedFullPath = styled(ChangeIndicatorWithProvidedFullPath)`
+  width: 1px;
+  height: 100%;
+
+  & > div {
+    height: 100%;
+  }
+`
+
+const InnerFlex = styled(Flex)`
+  position: relative;
+
+  [data-dragged] > & {
+    opacity: 0.5;
+  }
+`
+
+const BlockActionsOuter = styled(Box)`
+  width: 25px;
+  position: relative;
+
+  [data-dragged] & {
+    visibility: hidden;
+  }
+`
+
+const BlockActionsInner = styled(Flex)`
+  position: absolute;
+  right: 0;
+`
+
+const TooltipBox = styled(Box)`
+  max-width: 250px;
+`
+const BlockPreview = styled(Box)((props: {theme: Theme}) => {
+  const color = props.theme.sanity.color.input
+  return css`
+    background-color: ${color.default.enabled.bg};
+  `
+})
+
+export function BlockObject(props: BlockObjectProps) {
+  const {
+    attributes: {focused, selected, path},
+    block,
+    blockRef,
+    editor,
+    isFullscreen,
+    markers,
+    onChange,
+    onFocus,
+    readOnly,
+    renderBlockActions,
+    renderCustomMarkers,
+    type,
+  } = props
+  const elementRef = useRef<HTMLDivElement>()
+
+  const handleEdit = useCallback(() => {
+    onFocus(path.concat(FOCUS_TERMINATOR))
+  }, [onFocus, path])
+
+  const handleClickToOpen = useCallback(() => {
+    handleEdit()
+  }, [handleEdit])
+
+  const handleDelete = useCallback(() => {
+    PortableTextEditor.delete(
+      editor,
+      {focus: {path, offset: 0}, anchor: {path, offset: 0}},
+      {mode: 'block'}
+    )
+    PortableTextEditor.focus(editor)
+  }, [editor, path])
+
+  const blockPreview = useMemo(() => {
+    return (
+      <BlockObjectPreview
+        type={type}
+        value={block}
+        readOnly={readOnly}
+        onClickingDelete={handleDelete}
+        onClickingEdit={handleEdit}
+      />
+    )
+  }, [type, block, readOnly, handleDelete, handleEdit])
+
+  const tone = useMemo(() => {
+    if (selected || focused) {
+      return 'primary'
+    }
+
+    return 'default'
+  }, [focused, selected])
```

Comment:
It’s overkill to use `useMemo` here.
```suggestion
  const tone = selected || focused ? 'primary' : 'default'
```